src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -66,12 +77,13 @@
     <MakeDir Condition="'$(CoverageEnabledForProject)'=='true'" Directories="$(CoverageReportDir)" />
 
     <Exec Command="$(TestHost) $(TestCommandLine)"
-          WorkingDirectory="$(TestPath)%(TestTargetFramework.Folder)"
+          WorkingDirectory="@(TestTargetFrameworkPath)"
           ContinueOnError="true">
       <Output PropertyName="TestRunExitCode" TaskParameter="ExitCode" />
     </Exec>
 
     <Error Condition="'$(TestRunExitCode)' != '0'" Text="One or more tests failed while running tests from '$(MSBuildProjectName)' please check log for details!" />
+    <Touch Condition="'$(TestRunExitCode)' == '0'" Files="@(TestsSuccessfulSemaphore)" AlwaysCreate="true" />
   </Target>
```

Comment:
Perhaps we should just rename/move/copy the testresults.xml file to a testResults.failed.xml or testResults.succeeded.xml and use that as the semaphore.